### ISSUE TITLE
testsuite: cleanup list-jobs and tests that use hardcoded number for state

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#3433]
Per some discussion in #3428, I attempted revamping `t/job-manager/list-jobs` to output json instead of its tab delimited format.  That way we can use more modern jq commands to test, i.e. `jq .priority` instead of things like `cut -f5` which a new test writer would have no idea what that means or is outputting.

Conversion was going ok until a mountain of tests checked for a state string instead of a state number, i.e. `state=I` for inactive instead of `state=64`.

I think hard coding all those state numbers in would be worse than converting `list-jobs` to use json.  I could just change the `list-jobs` RPC in job-manager to return the state string, but figure that would be bad b/c it would be inconsistent to how `job-info.list` works.

A bit of discussion occurred in #2635 to create a `flux job statetostr` command or similar to convert the number of a string.

So I think good cleanup would be:

- write `flux job statetostr`, and i guess an option for abbreviations
- in many tests update to check state strings instead of numbers
  - this is also useful in case the state machine ever adds a state, we don't have to update all these hard coded numbers everywhere
- update list-jobs to use json instead of tab delimited output, update tests accordingly

**Comment by grondo** [comment]
why not rewrite `list-jobs` in Python, which can convert the ids to strings internally?

It would be cool if `list-jobs` output could be piped to `flux jobs --from-stdin`... not sure if the key names match what is expected in the `JobInfo` constructor though.

**Comment by grondo** [comment]
> It would be cool if list-jobs output could be piped to flux jobs --from-stdin...

Oh, hm, this works if you use a limited set of format fields!
```
$ flux python list-jobs.py | tail  | flux jobs --from-stdin -o "{id.f58:<8} {state:>8}"
JOBID       STATE
ƒJZsQKmX5 INACTIVE
ƒJa6SRTEw INACTIVE
ƒJi4Nyg4f INACTIVE
ƒJi4Nyg4g INACTIVE
ƒA6wibLc7 INACTIVE
ƒA6wibLc8 INACTIVE
ƒA6wibLc9 INACTIVE
ƒA6wibLcA INACTIVE
ƒA6wibLcB INACTIVE
ƒA6wibLcC INACTIVE
```

where `list-jobs.py` is
```python
import flux
import json

h = flux.Flux()
for job in h.rpc("job-manager.list", {"max_entries": 10}).get()["jobs"]:
    print(json.dumps(job))
```


BTW, `max_entries` doesn't seem to work in the `job-manager.list` protocol:

```
$ t/job-manager/list-jobs -c 10 | wc -l
14085
```

**Comment by chu11** [comment]
> why not rewrite list-jobs in Python, which can convert the ids to strings internally?

oh that's a good idea, lets try that

**Comment by chu11** [comment]
closed via #3436
